### ISSUE TITLE
Optional parameter followed by a hash parameter is not matched

### ISSUE THREAD

**Issue by urvashi01** [pillarjs/path-to-regexp#204]
For a path like 
`/something/:optional?#:hash`

an input of `/something#1` does not match in 1.7.0 and above.
It used to work in 0.1.7.

Strict, sensitive and end - all mode are false.

Is this expected?

**Comment by blakeembrey** [comment]
Unable to replicate, works on master:

```js
> require('.')('/something/:optional?#:hash').exec('/something#1')
[
  '/something#1',
  undefined,
  '1',
  index: 0,
  input: '/something#1',
  groups: undefined
]
```

**Comment by blakeembrey** [comment]
Confirmed that this was released ~11 months with 3.0: https://github.com/pillarjs/path-to-regexp/blob/master/History.md#300--2019-01-13.
